metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1452,6 +1453,16 @@ class MetalsLanguageServer(
       .ignoreValue
   }
 
+  private def onDelete(paths: Seq[AbsolutePath]): Future[Unit] = {
+    Future
+      .sequence(
+        List(
+          compilations.compileFiles(paths)
```

Comment:
Maybe you can move `diagnostics.didDelete` to here :)

```suggestion
          compilations.compileFiles(paths),
          diagnostics.didDelete(path)
```